src/contexts/KeyringContextProvider.tsx

```diff
@@ -39,16 +38,27 @@ export function KeyringContextProvider(props: KeyringContextProviderProps): Reac
   const [keyringStatus, setKeyringStatus] = useState(KeyringStatuses.INIT);
   const { dispatchMessage } = useUpdateMessageContext();
   const [keyringPairs, setKeyringPairs] = useState<Array<KeyringPair>>([]);
-  const [keyringPairsReady, setkeyringPairsReady] = useState(false);
+  const [keyringPairsReady, setkeyringPairsReady] = useState<boolean>(false);
+  const [extensionExists, setExtensionExists] = useState<boolean>(false);
+  const [accountExists, setAccountExists] = useState<boolean>(false);
 
   const isDevelopment = Boolean(process.env.REACT_APP_KEYRING_DEV_LOAD_ACCOUNTS);
 
   const loadAccounts = useCallback(() => {
     const asyncLoadAccounts = async () => {
       setKeyringStatus(KeyringStatuses.LOADING);
       try {
-        await web3Enable('Substrate Bridges UI');
+        const extExists = await web3Enable('Substrate Bridges UI');
+
+        if (extExists.length === 0) {
+          // no extension installed, or the user did not accept the authorization
+          // in this case we should inform the use and give a link to the extension
+          return;
+        } else {
+          setExtensionExists(true);
+        }
         let allAccounts = await web3Accounts();
+        allAccounts?.length > 0 && setAccountExists(true);
```

Comment:
minor: i think the `> 0` is not needed since lenght equal to zero would return false anyway